components/Base64Encode.coffee

```diff
@@ -5,55 +5,70 @@ unless noflo.isBrowser()
 else
   btoa = window.btoa
 
-class Base64Encode extends noflo.Component
-  description: 'This component receives strings or Buffers and sends them out
+exports.getComponent = ->
+  c = new noflo.Component
+  c.description = 'This component receives strings or Buffers and sends them out
   Base64-encoded'
 
-  constructor: ->
-    @data = null
-    @encodedData = ''
-
-    # This component has only two ports: an input port
-    # and an output port.
-    @inPorts = new noflo.InPorts
-      in:
-        datatype: 'all'
-        description: 'Buffer or string to encode'
-    @outPorts = new noflo.OutPorts
-      out:
-        datatype: 'string'
-        description: 'Encoded input'
-
-    # Initialize an empty string for receiving data
-    # when we get a connection
-    @inPorts.in.on 'connect', =>
-      @data = ''
-
-    # Process each incoming IP
-    @inPorts.in.on 'data', (data) =>
-      # In case of Buffers we can just encode them
-      # immediately
+  c.inPorts.add 'in',
+    datatype: 'all'
+    description: 'Buffer or string to encode'
+  c.outPorts.add 'out',
+    datatype: 'string'
+    description: 'Encoded input'
+
+  brackets = {}
+  scope = {}
+  rawData = {}
+  encodedData = {}
+
+  c.forwardBrackets = {}
+  c.process (input, output) ->
+    # Force auto-ordering to be off for this one
+    c.autoOrdering = false
+
+    return unless input.has 'in'
+    data = input.get 'in'
+    if data.type is 'openBracket'
```

Comment:
What's the reason bracket forwarding and auto-ordering doesn't work for this one? It seems like a simple thing at at glance: read `in`, encode, send to `out` - even synchronous. Do brackets have some sacral meaning here?